### PR TITLE
[redwood] Fix file permissions and add support for `build` script

### PR DESCRIPTION
**Pull request by styfle**
Some files require execution privileges, such as Prisma, so we must preserve the file mode.

We also want redwood to behave the same as other frameworks and use `yarn build` if available.

### CODE REVIEW

**Review by MaxLeiter — DISMISSED** [review]

**Comment by MaxLeiter** [comment]
Do we want to test this?

File: packages/redwood/src/index.ts [hunk 1]
```diff
@@ -119,9 +120,10 @@ export async function build({
       }),
     };
 
-    dependencies.forEach(fsPath => {
-      lambdaFiles[relative(workPath, fsPath)] = new FileFsRef({ fsPath });
-    });
+    for (const fsPath of dependencies) {
+      const { mode } = statSync(fsPath);
+      lambdaFiles[relative(workPath, fsPath)] = new FileFsRef({ fsPath, mode });
```
Comment (TooTallNate) [inline packages/redwood/src/index.ts]:
```suggestion
      lambdaFiles[relative(workPath, fsPath)] = await FileFsRef.fromFsPath({ fsPath });
```

File: packages/redwood/src/index.ts [hunk 2]
```diff
@@ -1,4 +1,5 @@
 import { join, dirname, relative, parse as parsePath, sep } from 'path';
+import { statSync } from 'fs';
```
Comment (TooTallNate) [inline packages/redwood/src/index.ts]: Can be removed with my suggestion [below](https://github.com/vercel/vercel/pull/4999/files#r466688471) (should be `lstat()` anyways 😉).

```suggestion
```

**Comment by TooTallNate** [comment]
Probably should add a test…

File: packages/redwood/src/index.ts (hunk 1 above)
Comment (styfle) [reply packages/redwood/src/index.ts]: Aha, I was wondering why the mode wasn't automatically set. This explains it 👍

**Review by TooTallNate — APPROVED** [review]

**Review by AndyBitz — APPROVED** [review]

**Pull request merged**